src/workload.rs

```diff
@@ -614,7 +639,10 @@ impl WorkloadStore {
                 }
             }
         }
-        if Some(&w.node) == self.local_node.as_ref() {
+        if Some(&w.node) == self.local_node.as_ref()
+            || (Some(&w.name) == self.local_pod.as_ref()
+                && Some(&w.namespace) == self.local_namespace.as_ref())
```

Comment:
Why do check workload name and namespace here?

Reply:
cc @howardjohn

Reply:
local_node `None` itself is very hack.

> fetch cert for current pod,

is this ztunnel itself?  Canot ztunnel fetch certs for those pods that run in same node (i mean in your case virtual kubelet)

Reply:
Yes, ztunnel share pod's name and namespace.

Reply:
> So in this case how could ztunnel be deployed, in only a single node or whole cluster as the daemonset

It's depends on the implementation. I think more fine-grained custom configurations may be needed to fine-grained control ztunnel behavior to be compatible with more serverless implementation scenarios.

Reply:
virtual kubelet cannot represent serverless

Reply:
We need a revist on the whole solution

Reply:
How about `ENABLE_IMPERSONATED_IDENTITY` only controls whether fetch cert with impersonated identity, and another env controls only fetch cert for specified pod if not empty.